File: previousJava/src/Enumerise/NutrientVitam.java
package Enumerise;

import java.util.HashMap;
import java.util.Map;

public enum NutrientVitam implements Nutrient{
	VITA("Vitamine A",0,"UI", UnitEnum.AUui,"VITA", "Vit A"),

	
	VITC("Vitamine C", 1, "mg",UnitEnum.BUmg,"VITC", "Vit C"),
	VITD("Vitamine D",2, "UI",UnitEnum.DUui,"VITD", "Vit D"),
	VITE("Vitamine E", 3,"UI",UnitEnum.EUui,"VITE", "Vit E"),
	VITK("Vitamine K", 4, "mg", UnitEnum.BUmg,"VITK", "Vit K"),
	VITB1("Thiamine (B1)", 5, "mg",UnitEnum.BUmg,"VITB1", "Vit B1"),
	VITB2("Riboflavine (B2)",6,"mg",UnitEnum.BUmg,"VITB2", "Vit B2"),
	VITB3("Nicotinamide, Niacine (B3/PP)",7,"mg",UnitEnum.BUmg, "VITB3", "Vit B3"),
	VITB5("Acide pantoth??nique (B5)",8,"mg",UnitEnum.BUmg,"VITB5", "Vit B5"),
	VITB6("Pyridoxine (B6)",9,"mg", UnitEnum.BUmg,"VITB6", "Vit B6"),
	VITB8("Biotine (B8)",10,"µg",UnitEnum.BUmu, "VITB8", "Vit B8"),
	VITB9("Acide folique (B9)",11,"µg",UnitEnum.BUmu, "VITB9", "Vit B9"),
	VITB12("Cyanocobalamine (B12)",12,"µg",UnitEnum.BUmu, "VITB12", "Vit B12"),
	CHOLINE("Choline", 13,"mg", UnitEnum.BUmg,"CHOLINE", "CHL"),

	RETINOL("Retinol",14,"µg", UnitEnum.BUmu,"RETINOL", "Ret"),
	BETACAR("B??ta-carot??ne",15,"µg", UnitEnum.BUmu,"BETACAR", "Bet")
;


private String name = "";
private int coef = 0;
private String unite="g";
private UnitEnum ue=UnitEnum.BUg;
private String label="VITA";
private String abr="Vit A";

//Constructeur
NutrientVitam(String name, int coef, String unite, UnitEnum ue,String label, String abr){
	this.name = name;
	this.coef=coef;
	this.unite=unite;
	this.ue=ue;
	this.label=label; 
	this.abr=abr;
}
public static boolean isByLabel(String label){
	
	for (NutrientVitam en:NutrientVitam.values()){
		if (en.getLabel().equals(label)){
			return true;
		}
	}
	return false;
}
public static NutrientVitam getByCoef(int i){
	
	return (NutrientVitam) map.get(i);
}
 private static Map map = new HashMap<>();
static {
    for (NutrientVitam pageType : NutrientVitam.values()) {
        map.put(pageType.coef, pageType);
    }
}
public static NutrientVitam getByLabel(String label){
	NutrientVitam enu=NutrientVitam.VITA;
	for (NutrientVitam en:NutrientVitam.values()){
		if (en.getLabel().equals(label)){
			enu=en;
		}
	}
	return enu;
}
public String getLabel() {
	return label;
}
public String getAbr() {
	return abr;
}
public String nameToString(){
	return name;
}
public String getUnite(){
	return unite;
}
public int getCoef(){
	return coef;
}
public UnitEnum getUe() {
	return ue;
}
public MainNutrientEnum getMNE() {
	return MainNutrientEnum.VITAM;
}
public static int size(){
	return 16;
}}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/NutrientVitam.kt
package fr.vetbrain.vetnutri_mp.Enumer

enum class NutrientVitam(
        val displayName: String,
        override val coef: Int,
        override val unite: String,
        override val ue: UnitEnum,
        val abr: String,
        override val label: String,
        /** Liste de labels alternatifs pour ce nutriment */
        val altLabels: List<String> = emptyList()
) : Nutrient {
        VITA("Vitamine A", 0, "UI", UnitEnum.AUui, "Vit A", "VITA", listOf("VIT_A", "VITAMINE_A")),
        VITC("Vitamine C", 1, "mg", UnitEnum.BUmg, "Vit C", "VITC", listOf("VIT_C", "VITAMINE_C")),
        VITD("Vitamine D", 2, "UI", UnitEnum.DUui, "Vit D", "VITD", listOf("VIT_D", "VITAMINE_D")),
        VITE("Vitamine E", 3, "UI", UnitEnum.EUui, "Vit E", "VITE", listOf("VIT_E", "VITAMINE_E")),
        VITK("Vitamine K", 4, "mg", UnitEnum.BUmg, "Vit K", "VITK", listOf("VIT_K", "VITAMINE_K")),
        VITB1(
                "Thiamine (B1)",
                5,
                "mg",
                UnitEnum.BUmg,
                "Vit B1",
                "VITB1",
                listOf("VIT_B1", "VITAMINE_B1")
        ),
        VITB2(
                "Riboflavine (B2)",
                6,
                "mg",
                UnitEnum.BUmg,
                "Vit B2",
                "VITB2",
                listOf("VIT_B2", "VITAMINE_B2")
        ),
        VITB3(
                "Nicotinamide, Niacine (B3/PP)",
                7,
                "mg",
                UnitEnum.BUmg,
                "Vit B3",
                "VITB3",
                listOf("VIT_B3", "VITAMINE_B3")
        ),
        VITB5(
                "Acide pantothénique (B5)",
                8,
                "mg",
                UnitEnum.BUmg,
                "Vit B5",
                "VITB5",
                listOf("VIT_B5", "VITAMINE_B5")
        ),
        VITB6(
                "Pyridoxine (B6)",
                9,
                "mg",
                UnitEnum.BUmg,
                "Vit B6",
                "VITB6",
                listOf("VIT_B6", "VITAMINE_B6")
        ),
        VITB8(
                "Biotine (B8)",
                10,
                "µg",
                UnitEnum.BUmu,
                "Vit B8",
                "VITB8",
                listOf("VIT_B8", "VITAMINE_B8")
        ),
        VITB9(
                "Acide folique (B9)",
                11,
                "µg",
                UnitEnum.BUmu,
                "Vit B9",
                "VITB9",
                listOf("VIT_B9", "VITAMINE_B9")
        ),
        VITB12(
                "Cyanocobalamine (B12)",
                12,
                "µg",
                UnitEnum.BUmu,
                "Vit B12",
                "VITB12",
                listOf("VIT_B12", "VITAMINE_B12")
        ),
        CHOLINE("Choline", 13, "mg", UnitEnum.BUmg, "CHL", "CHOLINE"),
        RETINOL("Retinol", 14, "µg", UnitEnum.BUmu, "Ret", "RETINOL"),
        BETACAR("Bêta-carotène", 15, "µg", UnitEnum.BUmu, "Bet", "BETACAR");

        override fun getMNE() = MainNutrientEnum.VITAM

        companion object {
                private val coefMap = values().associateBy { it.coef }
                private val labelMap = values().associateBy { it.label }

                /** Map des labels alternatifs vers leur nutriment correspondant */
                private val altLabelMap by lazy {
                        val map = mutableMapOf<String, NutrientVitam>()
                        values().forEach { nutrient ->
                                nutrient.altLabels.forEach { altLabel -> map[altLabel] = nutrient }
                        }
                        map
                }

                fun isByLabel(label: String): Boolean {
                        return label in labelMap || label in altLabelMap
                }

                fun getByCoef(coef: Int) = coefMap[coef]

                fun getByLabel(label: String): NutrientVitam? {
                        // Priorité aux labels originaux
                        return labelMap[label] ?: altLabelMap[label]
                }

                fun size() = values().size
        }
}
